crt_portal/cts_forms/views.py

```diff
@@ -50,11 +53,32 @@ def IndexView(request):
             p_class_list = p_class_list[:3]
             p_class_list[2] = f'{p_class_list[2]}...'
         data.append({
-            'report': report,
-            'report_protected_classes': p_class_list,
+            "report": report,
+            "report_protected_classes": p_class_list,
+            "url": '{}/'.format(report.id)
+        })
+
+    return render_to_response('forms/complaint_view/index.html', {
+        'data_dict': data,
+        'page_format': page_format,
+        'page_args': page_args,
+        'sort_state': sort
+    })
+
+
+@login_required
+def ShowView(request, id):
+    report = Report.objects.get(id=id)
```

Comment:
I think we might want to use get_object_or_404() here.

Reply:
I added a test for this, just wanted to note that the `test_required_user_logging` seems slightly brittle as written; it needs to be manually updated each time a new suite of tests is written involving an authenticated user. Perhaps there is another metric not tied to a specific number of database operations we could use for this test? I'm not familiar enough with the ins and outs of testing django apps quite yet, so I don't have any specific proposals, but I did want to flag it here. Happy to make an issue or document this elsewhere if it seems important!

Reply:
Generally, I create an object in the set up for the class, you can save the example record and then save `self.test_report_id = Report.pk` so you can call  reverse look up on `f'form/{self.test_report_id}'` in your tests and you know it exists.

Reply:
Maybe we could talk about this a bit more tomorrow, I don't think I'm explaining what I'm seeing correctly 😬